docs/guide/sorting-data.md

```diff
@@ -1,17 +1,202 @@
+
+1. First, HyperFormula sorts formulas by their results (calculated values).
+2. When the sorting is done, each formula is recalculated with regard to its new position.
+
+## Sorting rows
+
+To sort rows, use the [`isItPossibleToSetRowOrder`](../api/classes/hyperformula.md#isitpossibletosetroworder) and [`setRowOrder`](../api/classes/hyperformula.md#setroworder) methods.
+
+### Step 1: Choose a new row order
+Choose your required permutation of row numbers.
+
+For example, if you want to swap the first row with the third row, set the order to `[2, 1, 0]` instead of `[0, 1, 2]`:
+
+```js
+// a HyperFormula instance with example data
+const hfInstance = HyperFormula.buildFromArray([
+ [1],
+ [2],
+ [4, 5],
+]);
+
+// we'll set the row order to [2, 1, 0] in the next steps
+```
+
+::: tip
+The [`setRowOrder`](../api/classes/hyperformula.md#setroworder) method accepts an array of numbers, so you can implement any function that returns an array with your required row order.
+:::
+
+### Step 2: Check if the new row order can be applied
+
+Before you change the row order, check if your specified row number permutation can actually be applied.
+
+Thanks to the [`isItPossibleTo*` methods](basic-operations.md#isitpossibleto-methods), you can check if an operation is allowed, and display an error message if it's not.
+
+Use the [`isItPossibleToSetRowOrder`](../api/classes/hyperformula.md#isitpossibletosetroworder) method:
+
+```js
+const hfInstance = HyperFormula.buildFromArray([
+ [1],
+ [2],
+ [4, 5],
+]);
+
+// a variable to carry the user message
+let messageUsedInUI;
+
+// check if your permutation can be applied
+const isRowOrderOk = hfInstance.isItPossibleToSetRowOrder(0, [2, 1, 0]);
+
+// display an error message
+if (!isRowOrderOk) {
+  messageUsedInUI = 'Sorry, you cannot sort rows in this way.'
+}
+```
+
+### Step 3: Set the new row order
+
+If your specified row number permutation is valid, change the row order:
+
+```js
+const hfInstance = HyperFormula.buildFromArray([
+ [1],
+ [2],
+ [4, 5],
+]);
+
+let messageUsedInUI;
+
+const isRowOrderOk = hfInstance.isItPossibleToSetRowOrder(0, [2, 1, 0]);
+
+if (!isRowOrderOk) {
+  messageUsedInUI = 'Sorry, you cannot sort rows in this way.'
+} else {
+  // set the new row order
+  setRowOrder(0, [2, 1, 0]);
+}
+// rows 0 and 2 swap places
+
+// returns:
+// [{
+//   address: { sheet: 0, col: 0, row: 2 },
+//   newValue: 1,
+// },
+// {
+//   address: { sheet: 0, col: 1, row: 2 },
+//   newValue: null,
+// },
+// {
+//   address: { sheet: 0, col: 0, row: 0 },
+//   newValue: 4,
+// },
+// {
+//   address: { sheet: 0, col: 1, row: 0 },
+//   newValue: 5,
+// }]
+```
+
+## Sorting columns
+
+To sort columns, use the [`isItPossibleToSetColumnOrder`](../api/classes/hyperformula.md#isitpossibletosetcolumnorder) and [`setColumnOrder`](../api/classes/hyperformula.md#setcolumnorder) methods.
+
+### Step 1: Choose a new column order
+Choose your required permutation of column numbers.
```

Comment:
Maybe indexes or positions would be more precise?

Reply:
@wojciechczerniak - yup, I was thinking of indexes, but wasn't sure. Done: 9d9be0e